tiledb/sm/query/ast/query_ast.cc

```diff
@@ -82,15 +82,17 @@ Status ASTNodeVal::check_node_validity(const ArraySchema& array_schema) const {
     // Ensure that an attribute that is marked as nullable
     // corresponds to a type that is nullable.
     if ((!nullable) &&
-        (type != Datatype::STRING_ASCII && type != Datatype::CHAR)) {
+        (type != Datatype::CHAR && type != Datatype::STRING_ASCII &&
```

Comment:
Can we make a string_type function and replace everywhere?

Reply:
Are you thinking something like such?

https://github.com/TileDB-Inc/TileDB/pull/3766/commits/52594ea3a9a032247e4f91d8510b80ad0a02ec6d

Reply:
Yes

Reply:
Done.